src/Microsoft.DotNet.Darc/Darc/Options/UpdateSubscriptionCommandLineOptions.cs

```diff
@@ -36,6 +37,12 @@ internal class UpdateSubscriptionCommandLineOptions : CommandLineOptions
     [Option("failure-notification-tags", HelpText = "Semicolon-delineated list of GitHub tags to notify for dependency flow failures from this subscription")]
     public string FailureNotificationTags { get; set; }
 
+    [Option("source-enabled", HelpText = "Get only source-enabled (VMR code flow) subscriptions.")]
```

Comment:
Nullable bool command line option is logically inconsistent with expected command line behavior. Absent flag should be always equal to false flag. Introducing three state logic in command line options is violating the [Principle of Least Surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Reply:
I disagree.

> Absent flag should be always equal to false flag.

It should be equal to "do not change this field". For adding, it should be equal to "default" (there we could have non-nullable set to this default).
We have a subscription setting set to whatever value, but we want to change another setting only, like `FailureNotificationTags`. We need to denote somehow that we don't want to change `SourceEnabled` and that to me means `null`.

Or how would you handle this?

Reply:
I understand you have reasons for this, but the the logic itself doesn't matter in this case. It's about the command line option behaviors. Non-specified option should be always equal to option=false in behavior, otherwise it's confusing to the users and a bad UI behavior.

Reply:
But that's really not what we want. How would a user change a subscription's field? They would have to correctly supply all of the other values keeping them same as what is in the DB? That would be very error prone and undesirable?

Reply:
Imagine a random dev on the runtime team wanting to change the `FailureNotificationTags` field - it would be an impossible task?

Maybe how `darc` is designed and used doesn't conform with this but we'd have to rework all of this, making it very un-user-friendly in the process?

Reply:
Discussed online.